.github/workflows/release.yml

```diff
@@ -251,17 +251,58 @@ jobs:
     needs: [build, build-crosscompile, notarize-macos]
 
     steps:
+      - name: Checkout repository # we need package_index.template
+        uses: actions/checkout@v2
+
       - name: Download artifact
         uses: actions/download-artifact@v3
         with:
           name: ${{ env.ARTIFACT_NAME }}
           path: ${{ env.DIST_DIR }}
 
+      - name: Identify Prerelease
+        # This is a workaround while waiting for create-release action
+        # to implement auto pre-release based on tag
+        id: prerelease
+        run: |
+          wget -q -P /tmp https://github.com/fsaintjacques/semver-tool/archive/3.2.0.zip
+          unzip -p /tmp/3.2.0.zip semver-tool-3.2.0/src/semver >/tmp/semver && chmod +x /tmp/semver
+          if [[ "$(/tmp/semver get prerel "${GITHUB_REF/refs\/tags\//}")" ]]; then echo "::set-output name=IS_PRE::true"; fi
+
+      - name: Prepare artifacts for the release and handle package_index
+        run: |
+          package_index=`cat package_index.template | sed s/%%VERSION%%/${GITHUB_REF/refs\/tags\//}/`
+          declare -a target_folders=("Windows_32bit" "Windows_64bit" "Linux_64bit" "macOS_64bit" "Linux_32bit" "Linux_ARMv6" "Linux_ARM64")
+          cd dist
+          for folder in "${target_folders[@]}"
+          do
+            if [[ $folder = "Windows_32bit" || $folder = "Windows_64bit" ]]; then
+              ARCHIVE_NAME=${{ env.PROJECT_NAME }}_${GITHUB_REF/refs\/tags\//}_${folder}.zip
+            else
+              ARCHIVE_NAME=${{ env.PROJECT_NAME }}_${GITHUB_REF/refs\/tags\//}_${folder}.tar.gz
+            fi
+            T_OS=`echo ${folder} | awk '{print toupper($0)}'`
+            SHASUM=`sha256sum ${ARCHIVE_NAME} | cut -f1 -d" "`
+            SIZE=`stat --printf="%s" ${ARCHIVE_NAME}`
+            package_index=`echo $package_index |
+              sed s/%%FILENAME_${T_OS}%%/${ARCHIVE_NAME}/ |
+              sed s/%%FILENAME_${T_OS}%%/${ARCHIVE_NAME}/ |
+              sed s/%%SIZE_${T_OS}%%/${SIZE}/ |
+              sed s/%%SHA_${T_OS}%%/${SHASUM}/`
+          done
+          cd ..
+          echo ================== CUT ME HERE =====================
+          echo ${package_index}
+          echo ${package_index} > dist/package_index_draft.json
```

Comment:
```suggestion
          echo "${package_index}"
          echo "${package_index}" > dist/package_index_draft.json
```
Lack of quoting on the variable caused all line breaks to be lost from the package input snippet:

```json
{ "name": "imgtool", "version": "1.8.0-rc2", "systems": [ { "host": "i386-apple-darwin11", "url": "http://downloads.arduino.cc/tools/imgtool_1.8.0-rc2_macOS_64bit.tar.gz", "archiveFileName": "imgtool_1.8.0-rc2_macOS_64bit.tar.gz", "size": "8449569", "checksum": "SHA-256:7e8ea97ca6122898f274e18d0e1587451187e05a85d1afebd9f59d7b6fe9b337" }, { "host": "arm-linux-gnueabihf", "url": "http://downloads.arduino.cc/tools/imgtool_1.8.0-rc2_Linux_ARMv6.tar.gz", "archiveFileName": "imgtool_1.8.0-rc2_Linux_ARMv6.tar.gz", "size": "13471619", "checksum": "SHA-256:a73c489bde707ac2954d530bf163dfbc30cc40a8acb0a1bf85b05acfb02341e7" }, { "host": "aarch64-linux-gnu", "url": "http://downloads.arduino.cc/tools/imgtool_1.8.0-rc2_Linux_ARM64.tar.gz", "archiveFileName": "imgtool_1.8.0-rc2_Linux_ARM64.tar.gz", "size": "15648800", "checksum": "SHA-256:e4380c27b40197773cbd47bd89fee6d3597d72cc8a642f6a85f700db2df4f068" }, { "host": "x86_64-linux-gnu", "url": "http://downloads.arduino.cc/tools/imgtool_1.8.0-rc2_Linux_64bit.tar.gz", "archiveFileName": "imgtool_1.8.0-rc2_Linux_64bit.tar.gz", "size": "15574233", "checksum": "SHA-256:e1e6160d8c7f5ebf40a034f5a0572b86fb01fc81eb3b1277c7aebadc1712f722" }, { "host": "i686-linux-gnu", "url": "http://downloads.arduino.cc/tools/imgtool_1.8.0-rc2_Linux_32bit.tar.gz", "archiveFileName": "imgtool_1.8.0-rc2_Linux_32bit.tar.gz", "size": "13880766", "checksum": "SHA-256:2822d9ce5dc95afbe5e77d38ad8095282dd0d967af4cca6778d82ce0aa6ec586" }, { "host": "i686-mingw32", "url": "http://downloads.arduino.cc/tools/imgtool_1.8.0-rc2_Windows_32bit.zip", "archiveFileName": "imgtool_1.8.0-rc2_Windows_32bit.zip", "size": "6735783", "checksum": "SHA-256:8cfbf405393dd3ab383a0f32ea11ad145abe042fa10d1fcf76d0d1ef39c6ef11" }, { "host": "x86_64-mingw32", "url": "http://downloads.arduino.cc/tools/imgtool_1.8.0-rc2_Windows_64bit.zip", "archiveFileName": "imgtool_1.8.0-rc2_Windows_64bit.zip", "size": "8410907", "checksum": "SHA-256:4c07734f46c8f3957448839e231e89a15100efad2599b3faa00d3e2acc52461d" } ] }
```

Reply:
Ohhhh that's the reason behind that problem! Thanks a lot Per